backend/services/implementations/schoolService.ts

```diff
@@ -18,6 +18,33 @@ class SchoolService implements ISchoolService {
     this.userService = userService;
   }
 
+  // This method gets all schools from the database.
+  async getAllSchools(): Promise<SchoolResponseDTO[]> {
+    try {
+      const schools: Array<School> = await MgSchool.find();
+      return await Promise.all(
+        schools.map(async (school) => {
+          const teacherDTOs: Array<UserDTO> = await this.userService.findAllUsersByIds(
+            school.teachers,
+          );
+
+          return {
+            id: school.id,
+            name: school.name,
+            country: school.country,
+            subRegion: school.subRegion,
+            city: school.city,
+            address: school.address,
+            teachers: teacherDTOs,
+          };
+        }),
+      );
```

Comment:
thoughts on moving this logic to a helper function? I can see it being reused again in the future

Reply:
good idea :)